src/DeserializerGenerator.php

```diff
@@ -212,12 +212,12 @@ private function generateInnerCodeForFieldType(
                 return $this->generateCodeForArray($type, $arrayPath, $modelPropertyPath, $stack);
 
             case $type instanceof PropertyTypeDateTime:
-                if (null !== $type->getZone()) {
-                    throw new \RuntimeException('Timezone support is not implemented');
-                }
-                $format = $type->getDeserializeFormat() ?: $type->getFormat();
-                if (null !== $format) {
-                    return $this->templating->renderAssignDateTimeFromFormat($type->isImmutable(), (string) $modelPropertyPath, (string) $arrayPath, $format);
+                // todo: remove use of deprecated method {@link \Liip\MetadataParser\Metadata\PropertyTypeDateTime::getDeserializeFormat}
+                $formats = method_exists($type, 'getDeserializeFormats') ? $type->getDeserializeFormats() : null;
```

Comment:
instead of checking for method existence, we can change composer.json to require ^1.2.0 of metadata-parser.

Reply:
i think there would not be a BC break, because until now we would have errored if there were multiple formats, no? or i misunderstand, it would still be ok if we accept more valid but differently formatted dates than before, i don't see a problem there.

as long as we don't accidentally change the format when we serialize a date, i see no risks in adding more flexibility.